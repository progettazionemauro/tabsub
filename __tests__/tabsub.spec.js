import tabsub from '../src/index';

const dummy = () => {};

it('exports tabsub', () => {
  expect(typeof tabsub).toBe('function');
});

it('returns the proper API', () => {
  const radio = tabsub('new-channel');
  expect(radio).toMatchSnapshot();
});

it('works with smart defaults', () => {
  const radio = tabsub();
  expect(radio).toMatchSnapshot();
});

it('can work with different strategies', () => {
  const strategy = name => ({start: dummy, stop: dummy, close: dummy, post: dummy, on: dummy});
  const radio = tabsub('anotherOne', strategy);
  expect(radio.start).toBe(dummy);
  expect(radio.stop).toBe(dummy);
  expect(radio.close).toBe(dummy);
  expect(radio.post).toBe(dummy);
  expect(radio.on).toBe(dummy);
});
