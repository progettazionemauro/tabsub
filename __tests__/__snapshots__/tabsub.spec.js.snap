// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns the proper API 1`] = `
Object {
  "close": [Function],
  "name": "new-channel",
  "on": [Function],
  "post": [Function],
  "start": [Function],
  "stop": [Function],
}
`;

exports[`works with smart defaults 1`] = `
Object {
  "close": [Function],
  "name": "__DEFAULT_CHANNEL_",
  "on": [Function],
  "post": [Function],
  "start": [Function],
  "stop": [Function],
}
`;
